From ba380beb22563203168c96a9c13fa6e475252204 Mon Sep 17 00:00:00 2001
From: reichi <reichi@opendreambox.org>
Date: Tue, 21 Feb 2017 18:41:15 +0100
Subject: [PATCH 1/2] Menu: fix Menu icon loading failing on non-default skins
 when default has a matching svgj (which has the wrong path is and will
 therefore be ignored)

---
 usr/lib/enigma2/python/Screens/Menu.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/usr/lib/enigma2/python/Screens/Menu.py b/usr/lib/enigma2/python/Screens/Menu.py
index a6b21dd..b59af7d 100644
--- a/usr/lib/enigma2/python/Screens/Menu.py
+++ b/usr/lib/enigma2/python/Screens/Menu.py
@@ -10,7 +10,7 @@ from Components.PluginComponent import plugins
 from Components.config import config
 from Components.SystemInfo import SystemInfo
 
-from Tools.Directories import resolveFilename, SCOPE_SKIN, SCOPE_CURRENT_SKIN
+from Tools.Directories import resolveFilename, SCOPE_SKIN, SCOPE_CURRENT_SKIN, fileExists
 from Tools.LoadPixmap import LoadPixmap
 
 import xml.etree.cElementTree
@@ -32,13 +32,13 @@ def MenuEntryPixmap(entryID, png_cache, lastMenuID):
 	png = png_cache.get(entryID, None)
 	if png is None: # no cached entry
 		pngPath = resolveFilename(SCOPE_CURRENT_SKIN, "menu/" + entryID + ".svg")
-		if not pngPath:
-			resolveFilename(SCOPE_CURRENT_SKIN, "menu/" + entryID + ".png")
 		pos = config.skin.primary_skin.value.rfind('/')
 		if pos > -1:
 			current_skin = config.skin.primary_skin.value[:pos+1]
 		else:
 			current_skin = ""
+		if not fileExists(pngPath) or not (( current_skin in pngPath and current_skin ) or not current_skin ):
+			pngPath = resolveFilename(SCOPE_CURRENT_SKIN, "menu/" + entryID + ".png")
 		if ( current_skin in pngPath and current_skin ) or not current_skin:
 			png = LoadPixmap(pngPath, cached=True, size=pixmapSize) #lets look for a dedicated icon
 		if png is None: # no dedicated icon found
-- 
2.1.4

